### PR TITLE
Vite PWA For The Website 

### PR DESCRIPTION
**Pull request by Shinjan-saha**
#44 Issue Solved 

As Suggested used the Workbox Method for Caching Assets

### CODE REVIEW

**Comment by Shinjan-saha** [comment]
@tamalCodes Can you please review the PR

**Comment by tamalCodes** [comment]
> @tamalCodes Can you please review the PR

Sure, I was not well TBH. Also need some documentation - can you say what are the different strategies that are being used  here for caching ?

**Comment by Shinjan-saha** [comment]
> > @tamalCodes Can you please review the PR
> 
> Sure, I was not well TBH. Also need some documentation - can you say what are the different strategies that are being used here for caching ?

I used Vite-plugin & workbox approach for the caching of assets such as images

File: vite.config.js [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+          //     return url.pathname.startsWith("/api");
+          //   },
+          //   handler: "CacheFirst",
+          //   options: {
+          //     cacheName: "api-cache",
+          //     cacheableResponse: {
+          //       statuses: [0, 200],
+          //     },
+          //   },
+          // },
```
Comment (tamalCodes) [inline vite.config.js]: Remove these as of now

File: vite.config.js [hunk 2]
```diff
@@ -1,7 +1,62 @@
...
+            },
+          },
+          // Add more patterns for other asset types if needed
+          // {
+          //   urlPattern: /\.(css|js)$/i,
+          //   handler: "StaleWhileRevalidate",
+          //   options: {
+          //     cacheName: "css-js-cache",
+          //   },
+          // },
```
Comment (tamalCodes) [inline vite.config.js]: Also remove these as of now

**Review by tamalCodes — CHANGES_REQUESTED** [review]
The rest LGTM, let me do a couple of tests locally, I also have suggested a couple of changes

File: vite.config.js (hunk 1 above)
Comment (Shinjan-saha) [reply vite.config.js]: requested changes Done

File: vite.config.js (hunk 2 above)
Comment (Shinjan-saha) [reply vite.config.js]: requested changes done

Comment (tamalCodes) [reply vite.config.js]: Cool will be merging soon with local tests !

**Review by tamalCodes — APPROVED** [review]

**Pull request merged**